### PR TITLE
Remove all deprecated ignition API references

### PR DESCRIPTION
**Pull request by j-rivero**
The PR removes all the code that was added for the transition from ignition to gz name.

While I was opening this PR I thought that merging this at this point might complicate the possible forwards/backwards ports of other PRs from other branches :(

### CODE REVIEW

File: cmake/GzBenchmark.cmake [hunk 1]
```diff
@@ -59,15 +59,6 @@ function(_gz_add_version_info_target)
   )
 endfunction()
 
-function(ign_add_benchmarks)
-  # TODO(chapulina) Enable warnings after all libraries have migrated.
```
Comment (azeey) [inline cmake/GzBenchmark.cmake:63]: Note to self: we need to enable warnings in `gz-cmake3`.

File: examples/no_gz_prefix/README.md [hunk 2]
```diff
@@ -22,9 +22,9 @@ make package_source
 ## Configuring include directory names
 
 This package uses the `REPLACE_INCLUDE_PATH` option in `gz_configure_project`
-to allow a custom include path of `no_gz`, which doesn't start with `ignition/`.
-To confirm, build the package and observe that `AlmostEmpty.cc`
-compiles successfully while including `no_gz/Export.hh`:
+to allow a custom include path of `no_gz`. To confirm, build the package
```
Comment (azeey) [inline examples/no_gz_prefix/README.md:25]: Should this be
```suggestion
to allow a custom include path of `no_gz`, which doesn't start with `gz/`. To confirm, build the package 
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,5 @@
 # Gazebo CMake : CMake Modules for Gazebo Projects
 
-**Maintainer:** grey AT openrobotics.org
```
[the remark this answers is not in the document]
Comment (j-rivero) [reply README.md:3]: It was yes. I think that this comes from times when grey was the only maintainer of the library and not is a more team effort maintained library. We can list him as the original author if you prefer or list some other maintainers in here for completeness.

File: cmake/GzBenchmark.cmake (hunk 1 above)
Comment (j-rivero) [reply cmake/GzBenchmark.cmake:63]: https://github.com/gazebosim/gz-cmake/pull/346

**Review by azeey — APPROVED** [review]

**Pull request merged**